src/core/utils/chainProviders.ts

```diff
@@ -1,23 +1,104 @@
-import { ethers } from 'ethers'
+import { ethers, Contract } from 'ethers'
 import { Chain } from '../constants/chains'
+import { logger } from './logger'
+import { CustomMulticallJsonRpcProvider } from './CustomMulticallJsonRpcProvider'
 
-const provider = (url: string | undefined, chainId: Chain) =>
-  url ? new ethers.providers.StaticJsonRpcProvider(url, chainId) : undefined
+import { MulticallQueue } from './multicall'
+import multicallAbi from '../../contracts/abis/multicall.json'
+import { Multicall } from '../../contracts'
+import { MULTICALL_ADDRESS } from '../constants/MULTICALL_ADDRESS'
+
+const provider = ({
+  url,
+  chainId,
+  enableMulticall,
+}: {
+  url: string | undefined
+  chainId: Chain
+  enableMulticall: boolean
+}) => {
+  logger.info('Using multicall queue providers')
+  if (!url) return undefined
+
+  if (enableMulticall) {
```

Comment:
If multicall is set globally, we can just check `process.env.ENABLE_MULTICALL === 'true'` here and avoid having to pass the same parameter for every chain.

Reply:
My thinking is some chains might not have the multicall deployed yet

Reply:
so we have some control at a chain level